packages/realm-server/tests/indexing-test.ts

```diff
@@ -1024,6 +1004,167 @@ module(basename(__filename), function () {
+            } from 'https://cardstack.com/base/card-api';
+
+
+            export class Team extends CardDef {
+              static displayName = 'Team'
+              @field name = contains(StringField, {isUsed: true});
+            }
+
+            export class SimpleTask extends CardDef {
+              static displayName = 'Sprint Task';
+              @field team = linksTo(() => Team, {isUsed: true}); //even using isUsed doesn't fix this
+              @field shortId = contains(StringField, {
+                computeVia: function (this: Task) {
+                  return this.team?.name
+                },
+              });
+            }
+            `
+      );
+      await realm.write(
+        'team.json',
+        JSON.stringify(
+          {
+            "data": {
+              "type": "card",
+              "attributes": {
+                "name": "Team B",
+                "description": null,
+                "thumbnailURL": null
+              },
+              "meta": {
+                "adoptsFrom": {
+                  "module": "./task",
+                  "name": "Team"
+                }
+              }
+            }
+          }
+        )
+      );
+      await realm.write(
+        'task.json',
+        JSON.stringify(
+          {
+            "data": {
+              "type": "card",
+              "attributes": {
+                "title": null,
+                "description": null,
+                "thumbnailURL": null
+              },
+              "relationships": {
+                "team": {
+                  "links": {
+                    "self": "./team"
+                  }
+                }
+              },
+              "meta": {
+                "adoptsFrom": {
+                  "module": "./task",
+                  "name": "Task"
+                }
+              }
+            }
+          }
+
+        )
+      );
+
+      await realm.write(
+        'task-without-custom-isolated-template.json',
+        JSON.stringify(
+          {
+            "data": {
+              "type": "card",
+              "attributes": {
+                "title": null,
+                "description": null,
+                "thumbnailURL": null
+              },
+              "relationships": {
+                "team": {
+                  "links": {
+                    "self": "./team"
+                  }
+                }
+              },
+              "meta": {
+                "adoptsFrom": {
+                  "module": "./task-without-custom-isolated-template",
+                  "name": "SimpleTask"
+                }
+              }
+            }
+          }
+
+        )
+      );
+      let taskInstance = (await realm.realmIndexQueryEngine.instance(
+        new URL(`${testRealm}task`),
+      )) as any;
+      let taskInstanceWithoutCustomIsolatedTemplate = (await realm.realmIndexQueryEngine.instance(
+        new URL(`${testRealm}task-without-custom-isolated-template`),
+      )) as any;
```

Comment:
`taskInstance` is the failed case
`taskInstanceWithoutCustomIsolatedTemplate` is the success case